File: Practica5/Marco teorico/3. Destructor/D. Propiedades de los destructores/3.4.cpp
#include <iostream>
#include <string>


using namespace std;


class Ejemplo {
private:
    string nombre;


public:
    // Constructor
    Ejemplo(string n) : nombre(n) {
        cout << "Constructor llamado para " << nombre << endl;
    }


    // Destructor
    ~Ejemplo() {
        cout << "Destructor llamado para " << nombre << endl;
    }


    // Método para mostrar el nombre
    void mostrarNombre() {
        cout << "Nombre: " << nombre << endl;
    }
};

int main() {
    cout << "Creación de objetos:" << endl;
    Ejemplo objeto1("Objeto 1");
    Ejemplo objeto2("Objeto 2");


    cout << "\nMétodos llamados:" << endl;
    objeto1.mostrarNombre();
    objeto2.mostrarNombre();


    cout << "\nFin del ámbito de objeto2:" << endl;


    {
        Ejemplo objeto3("Objeto 3");
    } // Al llegar al final del ámbito, se llama al destructor de objeto3


    cout << "\nFin del ámbito de objeto1:" << endl;


    return 0;
}

File: Practica5/Marco teorico/3. Destructor/D. Propiedades de los destructores/3.4.java
public class Ejemplo {
    private String nombre;

    public Ejemplo(String n) {
        nombre = n;
        System.out.println("Constructor llamado para " + nombre);
    }

    public void mostrarNombre() {
        System.out.println("Nombre: " + nombre);
    }

    public static void main(String[] args) {
        System.out.println("Creación de objetos:");
        Ejemplo objeto1 = new Ejemplo("Objeto 1");
        Ejemplo objeto2 = new Ejemplo("Objeto 2");

        System.out.println("\nMétodos llamados:");
        objeto1.mostrarNombre();
        objeto2.mostrarNombre();

        System.out.println("\nFin del ámbito de objeto2:");

        {
            Ejemplo objeto3 = new Ejemplo("Objeto 3");
        } // Al llegar al final del ámbito, se llama al destructor de objeto3

        System.out.println("\nFin del ámbito de objeto1:");
    }
}
